#Glycine --> max, 
#Chromosome sizes in order: 
library(SimRAD)

#Glycine max chromosomes
GlyCh1.sq <- ref.DNAseq("~/Downloads/GlyMax.fasta")
GlyCh2.sq <- ref.DNAseq("~/Downloads/GmaxChr2.fasta")
GlyCh3.sq <- ref.DNAseq("~/Downloads/GmaxChr3.fasta")
GlyCh4.sq <- ref.DNAseq("~/Downloads/GmaxChr4.fasta")
GlyCh5.sq <- ref.DNAseq("~/Downloads/GmaxChr5.fasta")
GlyCh6.sq <- ref.DNAseq("~/Downloads/GmaxChr6.fasta")
GlyCh7.sq <- ref.DNAseq("~/Downloads/GmaxChr7.fasta")
GlyCh8.sq <- ref.DNAseq("~/Downloads/GmaxChr8.fasta")
GlyCh9.sq <- ref.DNAseq("~/Downloads/GmaxChr9.fasta")
GlyCh10.sq <- ref.DNAseq("~/Downloads/GmaxChr10.fasta")
GlyCh11.sq <- ref.DNAseq("~/Downloads/GmaxChr11.fasta")
GlyCh12.sq <- ref.DNAseq("~/Downloads/GmaxChr12.fasta")
GlyCh13.sq <- ref.DNAseq("~/Downloads/GmaxChr13.fasta")
GlyCh14.sq <- ref.DNAseq("~/Downloads/GmaxChr14.fasta")
GlyCh15.sq <- ref.DNAseq("~/Downloads/GmaxChr15.fasta")
GlyCh16.sq <- ref.DNAseq("~/Downloads/GmaxChr16.fasta")
GlyCh17.sq <- ref.DNAseq("~/Downloads/GmaxChr17.fasta")
GlyCh18.sq <- ref.DNAseq("~/Downloads/GmaxChr18.fasta")
GlyCh19.sq <- ref.DNAseq("~/Downloads/GmaxChr19.fasta")
GlyCh20.sq <- ref.DNAseq("~/Downloads/GmaxChr20.fasta")

#Restriction Enzyme 1
#EcoRI
cs_5p1 <- "G"
cs_3p1 <- "AATTC"
#Restriction Enzyme 2
#MseI
cs_5p2 <- "T"
cs_3p2 <- "TAA"


#Gen.dig <- insilico.digest(Glygen.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)

#digestion with EcoRI and MseI
GlyCh1.dig <- insilico.digest(GlyCh1.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh2.dig <- insilico.digest(GlyCh2.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh3.dig <- insilico.digest(GlyCh3.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh4.dig <- insilico.digest(GlyCh4.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh5.dig <- insilico.digest(GlyCh5.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh6.dig <- insilico.digest(GlyCh6.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh7.dig <- insilico.digest(GlyCh7.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh8.dig <- insilico.digest(GlyCh8.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh9.dig <- insilico.digest(GlyCh9.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh10.dig <- insilico.digest(GlyCh10.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh11.dig <- insilico.digest(GlyCh11.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh12.dig <- insilico.digest(GlyCh12.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh13.dig <- insilico.digest(GlyCh13.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh14.dig <- insilico.digest(GlyCh14.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh15.dig <- insilico.digest(GlyCh15.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh16.dig <- insilico.digest(GlyCh16.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh17.dig <- insilico.digest(GlyCh17.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh18.dig <- insilico.digest(GlyCh18.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh19.dig <- insilico.digest(GlyCh19.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)
GlyCh20.dig <- insilico.digest(GlyCh20.sq, cs_5p1, cs_3p1, cs_5p2, cs_3p2, verbose=TRUE)

#fragment type selection (with both cutsites)
GlyCh1.sel <- adapt.select(GlyCh1.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh2.sel <- adapt.select(GlyCh2.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh3.sel <- adapt.select(GlyCh3.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh4.sel <- adapt.select(GlyCh4.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh5.sel <- adapt.select(GlyCh5.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh6.sel <- adapt.select(GlyCh6.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh7.sel <- adapt.select(GlyCh7.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh8.sel <- adapt.select(GlyCh8.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh9.sel <- adapt.select(GlyCh9.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh10.sel <- adapt.select(GlyCh10.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh11.sel <- adapt.select(GlyCh11.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh12.sel <- adapt.select(GlyCh12.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh13.sel <- adapt.select(GlyCh13.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh14.sel <- adapt.select(GlyCh14.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh15.sel <- adapt.select(GlyCh15.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh16.sel <- adapt.select(GlyCh16.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh17.sel <- adapt.select(GlyCh17.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh18.sel <- adapt.select(GlyCh18.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh19.sel <- adapt.select(GlyCh19.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)
GlyCh20.sel <- adapt.select(GlyCh20.dig, type="AB+BA", cs_5p1, cs_3p1, cs_5p2, cs_3p2)

#wide size selection (200-400)
wid.Ch1 <- size.select(GlyCh1.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch2 <- size.select(GlyCh2.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch3 <- size.select(GlyCh3.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch4 <- size.select(GlyCh4.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch5 <- size.select(GlyCh5.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch6 <- size.select(GlyCh6.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch7 <- size.select(GlyCh7.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)
wid.Ch8 <- size.select(GlyCh8.sel, min.size = 200, max.size = 400, graph=TRUE, verbose=TRUE)

#narrow size selection (250-350)
nar.GCh1 <- size.select(GlyCh1.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh2 <- size.select(GlyCh2.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh3 <- size.select(GlyCh3.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh4 <- size.select(GlyCh4.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh5 <- size.select(GlyCh5.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh6 <- size.select(GlyCh6.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh7 <- size.select(GlyCh7.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh8 <- size.select(GlyCh8.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh9 <- size.select(GlyCh9.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh10 <- size.select(GlyCh10.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh11 <- size.select(GlyCh11.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh12 <- size.select(GlyCh12.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh13 <- size.select(GlyCh13.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh14 <- size.select(GlyCh14.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh15 <- size.select(GlyCh15.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh16 <- size.select(GlyCh16.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh17 <- size.select(GlyCh17.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh18 <- size.select(GlyCh18.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh19 <- size.select(GlyCh19.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)
nar.GCh20 <- size.select(GlyCh20.sel, min.size = 250, max.size = 350, graph=TRUE, verbose=TRUE)

narn.GCh1 <- size.select(GlyCh1.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh2 <- size.select(GlyCh2.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh3 <- size.select(GlyCh3.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh4 <- size.select(GlyCh4.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh5 <- size.select(GlyCh5.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh6 <- size.select(GlyCh6.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh7 <- size.select(GlyCh7.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh8 <- size.select(GlyCh8.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh9 <- size.select(GlyCh9.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh10 <- size.select(GlyCh10.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh11 <- size.select(GlyCh11.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh12 <- size.select(GlyCh12.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh13 <- size.select(GlyCh13.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh14 <- size.select(GlyCh14.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh15 <- size.select(GlyCh15.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh16 <- size.select(GlyCh16.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh17 <- size.select(GlyCh17.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh18 <- size.select(GlyCh18.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh19 <- size.select(GlyCh19.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)
narn.GCh20 <- size.select(GlyCh20.sel, min.size = 350, max.size = 450, graph=TRUE, verbose=TRUE)


#Exp. reads per lane / (number of samples * number of loci)

Nova <- 1300000000
MiSeq <- 22000000

MiCov <- MiSeq/(40*17000)
NovaCov <- Nova/(40*17000)
samples <- Nova/(60*17000)
